src/transformers/models/llama/modeling_llama.py

```diff
@@ -283,23 +283,37 @@ def __init__(self, dim, max_position_embeddings=2048, base=10000, device=None, s
         super().__init__(dim, max_position_embeddings, base, device)
 
     def _set_cos_sin_cache(self, seq_len, device, dtype):
+        if seq_len <= self.max_position_embeddings:
+            seq_len = self.max_position_embeddings
+
         self.max_seq_len_cached = seq_len
 
         if seq_len > self.max_position_embeddings:
             base = self.base * (
                 (self.scaling_factor * seq_len / self.max_position_embeddings) - (self.scaling_factor - 1)
             ) ** (self.dim / (self.dim - 2))
-            inv_freq = 1.0 / (base ** (torch.arange(0, self.dim, 2).float().to(device) / self.dim))
-            self.register_buffer("inv_freq", inv_freq, persistent=False)
+        else:
+            base = self.base
+        inv_freq = 1.0 / (base ** (torch.arange(0, self.dim, 2).float().to(device) / self.dim))
```

Comment:
The fact that this line is not within an if/else could explain part of the slowdown :)

(better to have some duplication than to have such a large slowdown)